test/app/adBlockTest.js

```diff
@@ -0,0 +1,27 @@
+/* global describe, it, before */
+
+const Brave = require('../lib/brave')
+const selectors = require('../lib/selectors')
+const messages = require('../../js/constants/messages')
+
+describe('adBlock', function () {
+  function * setup (client) {
+    yield client
+      .waitUntilWindowLoaded()
+      .waitForVisible('#window')
+      .waitForVisible(selectors.urlInput)
+  }
+
+  describe('blocks ruled ads', function () {
+    Brave.beforeAll(this)
+    before(function *() {
+      yield setup(this.app.client)
+    })
+
+    it('blocks a known rule on a local url', function *() {
+      yield this.app.client
+        .ipcSend(messages.SHORTCUT_NEW_FRAME, 'http://www.newsbtc.com/2016/02/14/bitcoin-price-weekly-analysis-primed-for-lift-off-2/')
+        .isExisting('.adfixedLeft').should.eventually.be.false
```

Comment:
each tab is actually its own window inside the main window. See "windw.open with click" for examples of interacting with the tab vs main window. Also, checking for non-existence is tricky and this would be subject to a race condition because "should.eventually.be" does not actually do what it sounds like it does. In the test/fixture that @bbondy mentioned you could create an element after dom load and check for the non-existence of the ad after that element is created.